### PR TITLE
Update README files for darktable 2.0

### PR DESCRIPTION
**Pull request by rwh86**

### CODE REVIEW

File: doc/README [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
- - gtk+-2.0 (and dependencies, e.g. cairo, gdkpixbuf, etc.)
- - png, jpeg, tiff libraries (should be pulled by gdkpixbuf)
- - exiv2, OpenEXR, lensfun, lcms2, curl
+http://redmine.darktable.org/projects/darktable/wiki/Building_darktable_20
+
+in summary, you'll need:
+
+ - gcc, g++, GNU make and cmake
+ - gtk+-3.0 (and dependencies, e.g. cairo, gdkpixbuf, etc.)
+ - xml, xslt and json libraries
```
Comment (LebedevRI) [inline doc/README]: +pugixml

File: doc/README [hunk 2]
```diff
@@ -14,22 +20,46 @@ optionally, you might need for special features:
  - flickcurl (for Flickr support)
  - libGL/sdl12 (to build slideshow viewer)
  - dbus-glib, gnome-keyring (kwallet/gnome keyring support)
```
Comment (LebedevRI) [inline doc/README]: gnome-keyring is gone, upstream fully deprecated it

File: doc/README [hunk 3]
```diff
@@ -14,22 +20,46 @@ optionally, you might need for special features:
...
+other used packages (supplied in the source tree):
+
+ - RawSpeed
+ - Lua 5.2 and LuaAutoc
+
+darktable has OpenCL support for cards with:
+
+ - > 1GB graphics RAM (more is better)
+ - a modern AMD or nVidia chipset
+ - proprietary drivers from the manufacturer loaded
```
Comment (LebedevRI) [inline doc/README]: proprietary drivers aren't exactly required
it's just that there is *still* no open opencl drivers that support all the stuff

**Comment by LebedevRI** [comment]
Random thought:

Oh, and while we are at it, maybe we should add "CONTRIBUTING.md" file that is automatically shown on "Create new PR" page.
Not sure we need it, but maybe something "have you talked to us first, before doing all this work?" :)

**Comment by rwh86** [comment]
OK, I'll see if I can pull some content from elsewhere into a CONTRIBUTING.md.

File: doc/README [hunk 4]
```diff
@@ -13,23 +19,49 @@ optionally, you might need for special features:
  - librsvg2 (for watermark plugin)
  - flickcurl (for Flickr support)
  - libGL/sdl12 (to build slideshow viewer)
- - dbus-glib, gnome-keyring (kwallet/gnome keyring support)
- - osm-gps-map for geo tagging view
+ - dbus-glib
```
Comment (houz) [inline doc/README]: We no longer use dbus-glib (since commit acf9329921e662efcc158211fe0654b3913ab6ca).

File: doc/README [hunk 5]
```diff
@@ -13,23 +19,49 @@ optionally, you might need for special features:
...
+ - osm-gps-map and libsoup for geo tagging view
  - lensfun (lens distortion plugin)
  - OpenEXR for HDR export
-
-other used packages (supplied in the source tree):
-
- - LibRaw nikon_curve (taken from ufraw)
- - RawSpeed
- - squish
+ - SDL, SDL-image and Mesa OpenGL for the slideshow view
```
Comment (houz) [inline doc/README]: That's for darktable-viewer, not slideshow view.

File: doc/README [hunk 6]
```diff
@@ -13,23 +19,49 @@ optionally, you might need for special features:
...
+$ evince doc/usermanual/darktable-usermanual.pdf
+
+optionally, to build translations of the manual pages:
+
+ - PO for anything (po4a)
+
+other used packages (supplied in the source tree):
+
+ - RawSpeed
+ - Lua 5.2 and LuaAutoc
```
Comment (houz) [inline doc/README]: Lua 5.2 can also be used from a system install. To use the copy we ship you have to tell cmake.

Comment (rwh86) [inline doc/README]: I found while I was testing the build on multiple platforms that I used the included Lua 5.2 more often than not, and I didn't have to tell cmake to do anything differently (I generally build with build.sh).

**Comment by simonspa** [comment]
Suggestion for additional CONTRIBUTING.md content:


# Getting Started
* make sure you have a user account on [GitHub](https://github.com/signup/free) and are logged in
* 'fork' the (main) project on GitHub [using the button on the page of the main repo](https://github.com/darktable-org/darktable/fork)

# General guidelines
* make atomic commits (aka commit often)
* for anything but trivial changes (e.g. affecting documentation only)
  please do not work on the ```master``` branch, but create a feature
  branch (see below for instructions) and commit only changes that
  affect this particular feature
* use descriptive commit messages -- for an example, see below
* follow the style (indentation, naming schemes for variables, ...) of the existing code, where possible

# Making Changes
Have you fixed a bug already or would like to contribute your own
processor into the main repository? Then please fork the project and
issue a pull request using these instructions:

* now add the newly forked repository as a git remote to your local darktable installation and rename the original repository to 'upstream':
```
git remote rename origin upstream
git remote add origin https://github.com/[YOUR GITHUB USER HERE]/darktable
git remote -v show
```
* create and checkout a new branch for the feature you are developing based on the ```master``` branch:
```
git branch my_feature_branch master
git checkout my_feature_branch
```

* now edit away on your local clone! But keep in sync with the development in the upstream repository by running
```
git pull upstream master
```
on a regular basis. Replace ```master``` by the appropriate branch if you work on a separate one *within the main repository*.

* Commit often using a descriptive commit message in the form of
```
This is a concrete and condensed headline for your commit

After the headline followed by an empty line, add a more thorough
description of what you have changed and why.  Refer to any issues on
GitHub affected by the commit by writing something along the lines of
e.g. 'this addresses issue darktable-org/darktable#1' to refer to the
first issue in the main repository.  
```

* push the edits to origin (your fork)
```
git push origin my_feature_branch
```

* verify that your changes made it to your GitHub fork and then click there on the 'compare & pull request' button

* summarize your changes and click on 'send'

* now we will review your changes and comment on anything unclear; any
  further commits you will make before the pull request is merged will
  also end up in the same pull request, so you can easily add or
  correct parts of your contribution while the review is ongoing.

**Comment by LebedevRI** [comment]
@simonspa: @boucman had a question about wordpress part of #1012

**Comment by rwh86** [comment]
@simonspa: where is that content from?  Googling suggests it's from [here](https://github.com/eutelescope/eutelescope/blob/master/CONTRIBUTING.md), which is AFAICT doesn't have an explicit open-source license, in fact I can't tell what license it's under.

**Comment by rwh86** [comment]
Closing this pull request as it was on my master branch.  Moved to its own branch, [new pull request](https://github.com/darktable-org/darktable/pull/1058).

**Pull request closed without merging**